spec/GameSpec.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+
+describe('Game', function() {
+  let game, frame, frame_1;
+
+  beforeEach(function(){
+    game = new Game();
+    frame = jasmine.createSpyObj('Frame',['frame_Score', 'AddRolls', 'isStrike'])
+    frame_1 = jasmine.createSpyObj('Frame',['frame_Score', 'AddRolls', 'isStrike'])
+  })
+
+  describe('#NextFrame', function() {
+    it('goes to the next frame after a strike', function(){
+      game.run(10);
+      expect(game.frame_counter).toEqual(1);
+    })
+
+    it('does not go to the next frame until finished', function(){
+      game.run(4);
+      expect(game.frame_counter).toEqual(0);
+    })
+
+    it('goes to the next frame after frame finishes', function(){
+      game.run(3);
+      game.run(3);
+      expect(game.frame_counter).toEqual(1);
+    })
+  })
+
+  describe('#AddStrikePoints', function() {
+    it('adds strike points to the previous frame', function() {
+      game.run(10);
+      game.run(3);
+      game.run(3);
+      expect(game.FrameBeforeLast().frame_score).toEqual(16);
+    })
+
+    it('does not add strike points to the previous frame', function() {
```

Comment:
why not? The two tests sound contradictory for someone who doesn't know in which cases this should happen. Describing the scenario in more detail could help here.

same for similar cases below